src/ompl/geometric/planners/rrt/src/RRTstar.cpp

```diff
@@ -632,14 +709,112 @@ void ompl::geometric::RRTstar::deleteBranch(Motion *motion)
     }
 }
 
-ompl::base::Cost ompl::geometric::RRTstar::costToGo(const Motion *motion, const bool shortest) const
+ompl::base::Cost ompl::geometric::RRTstar::defaultCostToGoHeuristic(const base::State *state, const base::Goal *goal) const
+{
+    if (pdef_->getGoal()->hasType(base::GOAL_STATE) == true)
+    {
+        return opt_->motionCost(state, goal->as<base::GoalState>()->getState());
+    }
+    else if (pdef_->getGoal()->hasType(base::GOAL_REGION) == true)
+    {
+        return base::goalRegionCostToGo(state, goal);
```

Comment:
My understanding from the documentation is that this function only works for `GOAL_REGION` types? Hence the switch. I guess I could handle the GOAL_STATE**S** case by finding the min over the countable set. I will update that when I push changes back from your comments. :smile: